### PR TITLE
Vishrut/build

### PR DESCRIPTION
**Pull request by vishrutsingh**
### Additions
- added a GitHub pre-commit hook for formatting
- added npm package files for publishing
- added type declarations for the project
- added release hooks and jobs

### Changes
- package.json dependencies: Build is relatively small now

### Missing
- README.md: can be pushed in the next PR

### CODE REVIEW

**Comment by vishrutsingh** [comment]
types directory is generated but required for build.

File: .github/workflows/RELEASE.yaml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+        with:
+          node-version: ${{ matrix.node-version }}
+      - name: Install dependencies
+        run: npm ci
+      - name: Build
+        run: npm run build
+      - name: Publish
+        run: npm publish
+        env:
+          NODE_AUTH_TOKEN: ${{ secrets.NPM_TOKEN }}
```
Comment (Anmol1696) [inline .github/workflows/RELEASE.yaml:28]: new line

File: .npmrc [hunk 2]
```diff
@@ -0,0 +1 @@
+scripts-prepend-node-path=true
```
Comment (Anmol1696) [inline .npmrc:1]: new line

File: package.json [hunk 3]
```diff
@@ -1,5 +1,5 @@
 {
-    "name": "persistencejs",
+    "name": "persistenceonejs",
```
Comment (Anmol1696) [inline package.json:2]: why the name change?

**Review by Anmol1696 — CHANGES_REQUESTED** [review]

**Review by Anmol1696 — APPROVED** [review]
Just nits, LGTM

**Comment by vishrutsingh** [comment]
have changed the scope of pre commit format to the whole repo for linting issues in directories other than src.

**Review by Anmol1696 — APPROVED** [review]
LGTM good work on running the linters

**Pull request merged**